File: src/main/java/com/redbox/domain/notice/dto/RecentNoticeResponse.java
package com.redbox.domain.notice.dto;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import lombok.Getter;

import java.time.LocalDate;

@Getter
@JsonDeserialize
public class RecentNoticeResponse {

    private Long noticeNo;
    private String title;
    private LocalDate createdDate;

    @JsonCreator
    public RecentNoticeResponse() {
    }

    public RecentNoticeResponse(Long noticeNo, String title, LocalDate createdDate) {
        this.noticeNo = noticeNo;
        this.title = title;
        this.createdDate = createdDate;
    }
}

File: migration/src/main/kotlin/com/redbox/domain/community/notice/dto/RecentNoticeResponse.kt
package com.redbox.domain.community.notice.dto

import com.fasterxml.jackson.annotation.JsonTypeInfo
import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import java.time.LocalDate

@JsonDeserialize
@JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, include = JsonTypeInfo.As.PROPERTY, property = "@class")
data class RecentNoticeResponse(
    val noticeNo: Long,
    val title: String,
    val createdDate: LocalDate
)
